### PR TITLE
perf(collaboration): add check permission for app studio

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/appstudio/index.ts [hunk 1]
```diff
@@ -278,6 +278,25 @@ export class AppStudioPlugin implements Plugin {
     return ok(localTeamsAppId);
   }
 
+  public async checkPermission(ctx: PluginContext): Promise<Result<any, FxError>> {
+    TelemetryUtils.init(ctx);
+    TelemetryUtils.sendStartEvent(TelemetryEventName.checkPermission);
+
+    try {
+      const checkPermissionResult = await this.appStudioPluginImpl.checkPermission(ctx);
+      TelemetryUtils.sendSuccessEvent(TelemetryEventName.checkPermission);
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/resource/appstudio/index.ts:287]: Do we need to add some properties for the telemetry?

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/appstudio/index.ts:287]: As commented in aad plugin, telemetry will be added after we have conclusion with yanliang in another pr. Here I only add least properties.

File: packages/fx-core/src/plugins/resource/appstudio/appStudio.ts [hunk 2]
```diff
@@ -135,9 +136,13 @@ export namespace AppStudioClient {
           `Cannot get the app definition with app ID ${teamsAppId}, due to ${e.name}: ${e.message}`
         );
       }
-      throw new Error(
+      const err = new Error(
         `Cannot get the app definition with app ID ${teamsAppId}, due to ${e.name}: ${e.message}`
       );
+      if (e.response?.data?.status) {
+        err.name = e.response?.data?.status;
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/resource/appstudio/appStudio.ts:143]: Use name is confusing, how about use statusCode since error is any type?

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/appstudio/appStudio.ts:143]: Error type only have name and message property. I am also considering whether it is better to add a new interface here.

**Review by nliu-ms — APPROVED** [review]

**Review by adashen — APPROVED** [review]

**Pull request merged**